### ISSUE TITLE
Brace throws error

### ISSUE THREAD

**Issue by efender** [shannonmoeller/gulp-hb#33]
If project path has parenthesis than the plugin (v. 4, but not v.2.6.5) throws error, e.g. "(folder) name":
```bash
# ./(folder) name
# ./(folder) name/gulpfile.js
events.js:154
      throw er; // Unhandled 'error' event
      ^
Error: Missing partial: 'layouts/base' ...
```
But without parenthesis ("folder name") it's ok

P.S. Thanks for your great plugin )

update: sorry for my english, not "braces", but "parenthesis" )

**Comment by shannonmoeller** [comment]
I think it might be that the partial names are different now. The way partials are named is a bit different than it used to be. See the section [Exporting a Function](https://github.com/shannonmoeller/handlebars-wax#exporting-a-function) in the [`handlebars-wax`](https://github.com/shannonmoeller/handlebars-wax) README.

Try adding the `debug` flag to your options and double check that your partials are registered as you expect:

```js
hb({ debug: true })
```

**Comment by efender** [comment]
But without braces it works!
Little more error log:
```bash
Error: Missing partial: 'layouts/base'
    at Object.helpers.extend (/path-to-project/(test)/node_modules/handlebars-layouts/index.js:122:11)
```
I append console.log at handlebars-layouts/index.js:
![1455889305111](https://cloud.githubusercontent.com/assets/6908417/13177047/bc00ff30-d727-11e5-8197-6eb2a61350c4.png)
Log for project "(test)" (with braces):
```bash
{ '(test)/src/view/partials/blocks/footer': { [Function: ret] _setup: [Function], _child: [Function] },
  '(test)/src/view/partials/blocks/head': { [Function: ret] _setup: [Function], _child: [Function] },
...
  '(test)/src/view/partials/layouts/base': { [Function: ret] _setup: [Function], _child: [Function] } }
```

Log for project "test" (no braces):
```bash
{ 'blocks/footer': { [Function: ret] _setup: [Function], _child: [Function] },
  'blocks/head': { [Function: ret] _setup: [Function], _child: [Function] },
...
  'layouts/base': { [Function: ret] _setup: [Function], _child: [Function] } }
```

One task, one project (only project name with and w/o ()) but different return:
**(test)/src/view/partials/**`blocks/footer` vs '`blocks/footer`

It may help?

---
And debug log is very strange:
![1455889776675](https://cloud.githubusercontent.com/assets/6908417/13177240/d92b7b70-d728-11e5-85d6-da234b93da36.png)

My gulp task:
```javascript
import c from '../config';
import {
    gulp, data, fm, hb, rename
} from '../imports';

export default function handlebarsTask() {
    const hbStream = hb(
        {
            debug: true,
            bustCache: true
        } )
        .partials( c.src + '/view/partials/**/*.hbs' )
        .helpers( 'node_modules/handlebars-layouts/index.js' )
        .helpers( c.src + '/view/helpers/**/*.js' )
        .data( c.src + '/view/data/**/*.{js,json}' );

    return gulp
        .src( c.src + '/view/pages/**/*.hbs' )
        .pipe( data( function ( file ) {
            return require( file.path.replace( '.html', '.json' ) );
        } ) )
        .pipe( fm( { property: 'data.fm' } ) )
        .pipe( hbStream )
        .pipe( rename( function ( path ) {
            path.extname = '.html';
            return path;
        } ) )
        .pipe( gulp.dest( c.dest ) );
}
```

**Comment by shannonmoeller** [comment]
Yep. This is the expected behavior. Parenthesis are considered glob characters (see [`node-glob`: Glob Primer](https://github.com/isaacs/node-glob#glob-primer). Partial names are derived from the [non-glob portion](https://github.com/shannonmoeller/handlebars-wax#exporting-a-function) of a given path using [`glob-parent`](https://github.com/es128/glob-parent#usage).

```js
// Without the parenthesis
.partials('test/view/partials/**/*.hbs');
//                            └─ glob starts here

// With the parenthesis
.partials('(test)/view/partials/**/*.hbs');
//         └─ glob starts here
```

The debug output is long because it looks like the debug flag is passed all the way down to `node-glob`. I'll look into that. The `gulp-hb`-specific portion is at the bottom and displays what you're logging:

```
  main.hbs
    context:
      settings
    data:
      fm
    decorators:
      inline
    helpers:
      add
      ...
      with
    partials:
      blocks/footer
      blocks/head
      blocks/header
      blocks/popups
      blocks/scripts
      blocks/sidebar
      blocks/vk_widget
      layouts/base
```

**Comment by shannonmoeller** [comment]
In v2, partials were named according to the unique portion of a path, regardless of where the globbing happened. While this worked as expected in certain cases, it led to cases where paths would change in unexpected ways simply by creating new partials:

```
test/view/partials/layouts/foo.hbs
test/view/partials/layouts/bar.hbs
                           └─ unique portion starts here

// ... add some partials ...

test/view/partials/components/foo.hbs
test/view/partials/components/bar.hbs
test/view/partials/layouts/foo.hbs
test/view/partials/layouts/bar.hbs
                   └─ unique portion now starts here
```

Standardizing on globbed portions of paths and introducing the fluent API gave users the ability to predictably and granularly control exact names of data keys, partials, helpers, and decorators.

**Comment by efender** [comment]
So, I can't use parenthesis in folder names anymore?
I should revert back to v.2? )

**Comment by shannonmoeller** [comment]
In general, I would recommend against using parenthesis in paths in favor of an alternate pattern. Perhaps `_test` instead of `(test)`? Because this is core glob syntax, I don't plan on changing the new behavior.

If you really wish to continue using parenthesis, you may provide your own [`parsePartialName` callback](https://github.com/shannonmoeller/handlebars-wax#partialspattern--options-handlebarswax). The original logic can be found in the [v1 branch of `require-glob`](https://github.com/shannonmoeller/require-glob/blob/v1/index.js#L8-L43).

**Comment by efender** [comment]
Ок, thanks!

**Comment by shannonmoeller** [comment]
My pleasure!

**Comment by shannonmoeller** [comment]
@efender FYI, I've addressed the issue with `debug` being too noisy and released as [v4.0.3](https://github.com/shannonmoeller/gulp-hb/releases/tag/v4.0.3).

**Comment by efender** [comment]
Yes, now debug output is clean )
Thanks!

P.S. I replaced parentheses by equals `(name1) name2` -> `=name1= name2`.
